### PR TITLE
Run and fail tests when ITestDataSource.GetData returns empty IEnumerable

### LINKED ISSUE #1052: When data source provides no data, it is impossible to see why tests does not run
## Description
When data source provides no data it is very hard to understand why test does not run. There is no warning that can be easily seen. And debugger breakpoint in the data source class is not hit (because it only attaches in run). 

## Steps to reproduce
```cs
using FluentAssertions;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Reflection;

namespace mstest44
{
    [TestClass]
    public class UnitTest1
    {
        [TestMethod]
        [MyDataSource(1)]

        public void TestMethod1(int i)
        {
            
        }

        [TestMethod]
        [MyDataSource(0)]

        public void TestMethod2(int i)
        {

        }
    }

    public class MyDataSource : Attribute, ITestDataSource
    {
        private readonly int _count;

        public MyDataSource(int count)
        {
            _count = count;
        }

        public IEnumerable<object[]> GetData(MethodInfo methodInfo)
        {
            var items = new List<object[]>();
            foreach (var i in Enumerable.Range(100, _count))
            {
                items.Add(new object[] { i });
            }

            return items;
        }

        public string GetDisplayName(MethodInfo methodInfo, object[] data)
        {
            return null;
        }
    }
}
```

## Expected behavior
I expect to see some warning, but have the ability to force not generating any data. Probably as an additional property on the data source interface?

## Actual behavior

One of my data sources are is not running any tests, because I did not add the data to the collection, and the test stays blue in TE, and I can't figure it out. 

image

## Environment
Latest TP, with latest mstest.

AB#2051847

### PR DESCRIPTION
**Pull request by fhnaseer**
Fixes #1052 and follows #1748

### CODE REVIEW

File: test/IntegrationTests/MSTest.Acceptance.IntegrationTests/DynamicDataTests.cs [hunk 1]
```diff
@@ -11,7 +11,8 @@ namespace MSTest.Acceptance.IntegrationTests;
 public class DynamicDataTests : AcceptanceTestBase
```
Comment (Evangelink) [inline test/IntegrationTests/MSTest.Acceptance.IntegrationTests/DynamicDataTests.cs]: Can you please rename the file/class into `ParameterizedTestTests` or ` DataTestTests`? You are now testing more than just `DynamicData`.

File: src/Adapter/MSTest.TestAdapter/Discovery/AssemblyEnumerator.cs [hunk 2]
```diff
@@ -302,6 +302,11 @@ private static bool TryProcessTestDataSourceTests(UnitTestElement test, TestMeth
             try
             {
                 data = dataSource.GetData(methodInfo);
+
+                if (!data.Any())
```
Comment (Evangelink) [inline src/Adapter/MSTest.TestAdapter/Discovery/AssemblyEnumerator.cs:309]: Could you please add a comment to mention the other place to look at/change? And mention this one on the `TstMethodRunner`.

**Review by Evangelink — APPROVED** [review]
Good job @fhnaseer, I have added 3 small nits and we can merge after.

File: test/IntegrationTests/MSTest.Acceptance.IntegrationTests/DynamicDataTests.cs (hunk 1 above)
Comment (fhnaseer) [reply test/IntegrationTests/MSTest.Acceptance.IntegrationTests/DynamicDataTests.cs]: Updated,

File: src/Adapter/MSTest.TestAdapter/Discovery/AssemblyEnumerator.cs (hunk 2 above)
Comment (fhnaseer) [reply src/Adapter/MSTest.TestAdapter/Discovery/AssemblyEnumerator.cs:309]: Added,

**Review by Evangelink — APPROVED** [review]

**Pull request merged**